### ISSUE TITLE
Figure out how to parse dot config from Caddyfile

### ISSUE THREAD

**Issue by infogulch** [infogulch/xtemplate-caddy#5]
Currently xtemplate-caddy cannot parse dot config from a Caddyfile.

It would be nice if it could.

If not, at least test that using caddy json to configure dot providers works correctly, and write some examples to the readme.

**Comment by gedw99** [comment]
This would really help get things moving .

Will try to help where I can.

I have built a makefile to bring together all the things a dev needs to do to compile and run things. If you want it as a or etc just let me know ..

**Comment by infogulch** [comment]
I'm working on this issue in the [next branch of the xtemplate repo](https://github.com/infogulch/xtemplate/compare/master...next) that will make it possible to add support for Caddyfile-based dot config in xtemplate-caddy.

My diagnosis is that I tried too hard to break xtemplate up into optional build-time dependencies, which forced me to build an over-generic configuration system since xtemplate couldn't know what plugins you compiled in. Now I think the right solution is to hard-code dependencies to all the functionality I want it to have and add configuration options for each of them directly in the main Config struct.

The one thing I got out of this over-generic config system is that I don't want just *one* config option for each type, I want a *list* for each. This would allow you to configure e.g. multiple sql connections, or multiple nats connections, and customize their names. I don't think this will be very common, but it seems silly to not allow for it. Sometimes two steps forward then one step back is where you want to be. 😅 

It's not passing all tests yet. Once it's working and released in xtemplate I will be able to update xtemplate-caddy in turn. This will break existing configurations, but the template code should be unaffected.

**Comment by infogulch** [comment]
Not really, if you want multi-tenant where the client controls the nats org then you should probably use multiple instances of xtemplate with different configurations, and if you want multi-tenancy across a single nats org managed by you then you'd want to use different subject name segments. Since nats can integrate directly across users, accounts, and even orgs (I think) at the level of server-configuration there is less justification for multiple clients than SQL where, for example, you might want to access both OLTP and OLAP type databases or a local SQLite used as a cache etc.

Still, allowing multiple clients could be handy if you're trying to use an externally-managed nats service where they refuse to integrate at the server level. I'd rather have it and not need it and need it and not have it, especially when the total cost of allowing multiple is a single loop.

**Comment by infogulch** [comment]
Yes, it's worth a closer look but I think this is complete.
